### PR TITLE
Bring in the `winston` package for logging.

### PR DESCRIPTION
**Pull request by SeanCline**
In case you are looking for contributions for issue #649, here is a preliminary take at logging support. I'm not sure if this is what you were thinking in terms of logging. If not, I'll close the PR.

(Mostly, I just wanted to mess around with winston for logging since I'd never used it.)

### CODE REVIEW

**Comment by MarshallOfSound** [comment]
I only took a quick glance, need to sleep soon.  Couple of comments but overall pretty much EXACTLY :+1:  what I was planning on doing :smile:

File: src/inject/GPMInject/index.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const waitForExternal = setInterval(() => {
       try {
         fn();
       } catch (e) {
-        console.error(e); // eslint-disable-line
+        Logger.error('Emitter fn() threw exception.', e);
```
Comment (MarshallOfSound) [inline src/inject/GPMInject/index.js]: Anything inside "inject" won't have the global logger availible, electron works in two processes (renderer and main), look into this to figure out how you would make them global in both.

There is also a trick in electron that would go something like `require('electron').remote.getGlobal('globalNameHere')`

File: src/inject/GPMInject/interface/voiceControls/SpeechRecognizer.js [hunk 2]
```diff
@@ -95,7 +95,7 @@ export default class SpeechRecognizer {
 
     _.forEach(actions, (actionString) => {
       if (this._handlers[actionString.toLowerCase()]) {
-        console.error(`The action "${action.toLowerCase()}" already has a handler registered`); // eslint-disable-line
+        Logger.error(`The action "${action.toLowerCase()}" already has a handler registered`); // eslint-disable-line
```
Comment (MarshallOfSound) [inline src/inject/GPMInject/interface/voiceControls/SpeechRecognizer.js]: I would want this to go to the console aswell, not sure if that is how winston works or not

File: src/main/constants.js [hunk 3]
```diff
@@ -3,9 +3,5 @@ export const LASTFM_API_SECRET = process.env.GPMDP_LASTFM_API_SECRET;
 
 if (!LASTFM_API_KEY || !LASTFM_API_SECRET ||
   LASTFM_API_KEY === 'undefined' || LASTFM_API_SECRET === 'undefined') {
-  console.error('####################################'); // eslint-disable-line
-  console.error('############### ERROR ##############'); // eslint-disable-line
-  console.error('######### LASTFM  API KEYS #########'); // eslint-disable-line
-  console.error('######## HAVE NOT BEEN SET #########'); // eslint-disable-line
-  console.error('#####################################'); // eslint-disable-line
+  Logger.error('Last.fm API keys have not been set!');
```
Comment (MarshallOfSound) [inline src/main/constants.js]: I would want this to go to the console aswell, not sure if that is how winston works or not

File: src/main/features/core/autoUpdater.js [hunk 4]
```diff
@@ -57,7 +64,7 @@ const checkUpdateServer = () => {
   https.get('https://update.gpmdp.xyz', () => {
     setUpAutoUpdate();
   }).on('error', () => {
-    console.log('################### !! Update server down !! ##################'); // eslint-disable-line
+    Logger.error('################### !! Update server down !! ##################');
```
Comment (MarshallOfSound) [inline src/main/features/core/autoUpdater.js]: I would want this to go to the console aswell, not sure if that is how winston works or not

**Comment by SeanCline** [comment]
I'll address the issue with the global logger not being available in the renderer process later today.

Winston lets you select how verbose you want each of the transports to be. I chose to make it so only errors go to console and anything info level or higher goes to log. And, in dev mode, debug or higher go to console. This is very much open to tweaking.

**Pull request merged**